tests/baselines/reference/complexRecursiveCollections.errors.txt

```diff
@@ -1,3 +1,6 @@
+tests/cases/compiler/immutable.ts(189,20): error TS2430: Interface 'Stack<T>' incorrectly extends interface 'Indexed<T>'.
+  The types returned by 'concat(...).map(...).filter(...)' are incompatible between these types.
+    Type 'Set<any>' is not assignable to type 'Indexed<any>'.

```

Comment:
This is just a new follow-on error to the existing errors below, I believe, as the incorrect extensions below make `Set` and `Indexed` incompatible with one another, which is now properly reported here (as we now no longer fully erase all the data in the signature and can keep comparing until we get to that point).